packages/react/src/wrap.tsx

```diff
@@ -146,7 +147,7 @@ export const withErrorBoundaryGroup = <TProps extends ComponentProps<ComponentTy
 export const withAsyncBoundary = Object.assign(
   <TProps extends ComponentProps<ComponentType> = Record<string, never>>(
     Component: ComponentType<TProps>,
-    asyncBoundaryProps: PropsWithoutChildren<AsyncBoundaryProps>
+    asyncBoundaryProps: OmitKeyOf<AsyncBoundaryProps, 'children'>
```

Comment:
Easy to read and in addition We can guarantee that AsyncBoundaryProps have children prop